designs/metrics.md

```diff
@@ -0,0 +1,304 @@
+
+```go
+package instance
+
+var (
+    VMCreateStartMetric             // Tracks operations starting
+    VMCreateSyncFailureMetric       // Tracks synchronous failures
+    VMCreateAsyncFailureMetric      // Tracks asynchronous failures
+    VMCreateResponseErrorMetric     // Tracks response errors
+)
+```
+
+### 2. Updated Component Implementation
+
+**File Modified:** `pkg/providers/instance/instance.go`
+
+The component uses its co-located metrics directly, without needing centralized metric imports:
+
+**Before:**
+```go
+import "github.com/Azure/karpenter-provider-azure/pkg/metrics"
+
+metrics.ComponentMetric.Inc(ctx, "description", metrics.Label(value))
+```
+
+**After:**
+```go
+// No separate metrics import needed - metrics are in same package
+
+ComponentMetric.WithLabelValues(value).Inc()
+```
+
+## Design Principles
+
+### Metric Namespace Identification
+
+To distinguish Azure provider metrics from core Karpenter metrics, we evaluated using a provider-specific namespace but ultimately decided to **keep emitting metrics in the core `karpenter_*` namespace** for consistency.
+
+This is configured in `pkg/metrics/constants.go`:
+
+```go
+const (
+    Namespace = "karpenter"  // Reuse core namespace for consistency
+)
+```
+
+**Why This Matters:**
+- **Clear expectations** for consumers already relying on the core namespace
+- **Avoids conflicts** with upstream Karpenter metric names
+- **Keeps dashboards/alerts working** without additional filtering
+- **Consistent naming** across all components as we upstream more features
+
+**Metric Naming Examples:**
+- `karpenter_instance_vm_create_start_total` - Azure provider metric (new)
+- `karpenter_nodeclaims_created_total` - Core Karpenter metric
+
+### Namespace Rationale
+
+We are intentionally keeping provider metrics in the core `karpenter_*` namespace so that:
+
+- **Compatibility with upstream tooling** remains intact—existing dashboards and alerts maintained by the Karpenter community continue to function. For example, our controller still emits core metrics by wrapping the provider implementation with `metrics.Decorate` in `cmd/controller/main.go` (see [`cmd/controller/main.go#L63`](https://github.com/Azure/karpenter-provider-azure/blob/main/cmd/controller/main.go#L63)).
+- **Stability expectations** stay aligned—metrics such as `ProvisioningDurationSeconds` in `pkg/cloudprovider/cloudprovider.go` (see [`cloudprovider.go#L224`](https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/cloudprovider/cloudprovider.go#L224)) already ship in this namespace and have consumers.
+
+In practice, the controller will continue to emit the decorated core metrics for compatibility, and new Azure-specific metrics (such as the VM create counters) will also use the `karpenter` namespace to maintain a single, consistent surface area. This keeps expectations aligned with upstream Karpenter and simplifies future maintenance and documentation.
```

Comment:
minor: Worth clarifying that we'll have our own "subsystems" so it should still be relatively clear which metrics are from core versus us (plus we can document that too)